python/extensions.bzl

```diff
@@ -19,23 +19,48 @@ load("@rules_python//python/pip_install:pip_repository.bzl", "locked_requirement
 load("@rules_python//python/pip_install:repositories.bzl", "pip_install_dependencies")
 load("@rules_python//python/pip_install:requirements_parser.bzl", parse_requirements = "parse")
 load("@rules_python//python/private:coverage_deps.bzl", "install_coverage_deps")
+load("@rules_python//python/private:toolchains_repo.bzl", "get_host_os_arch", "get_host_platform")
 
 def _python_impl(module_ctx):
     for mod in module_ctx.modules:
-        for attr in mod.tags.toolchain:
+        for toolchain_attr in mod.tags.toolchain:
             python_register_toolchains(
-                name = attr.name,
-                python_version = attr.python_version,
+                name = toolchain_attr.name,
+                python_version = toolchain_attr.python_version,
                 bzlmod = True,
                 # Toolchain registration in bzlmod is done in MODULE file
                 register_toolchains = False,
-                register_coverage_tool = attr.configure_coverage_tool,
-                ignore_root_user_error = attr.ignore_root_user_error,
+                register_coverage_tool = toolchain_attr.configure_coverage_tool,
+                ignore_root_user_error = toolchain_attr.ignore_root_user_error,
+            )
+        for interpret_attr in mod.tags.host_python_interpreter:
+            _host_hub(
+                name = interpret_attr.name,
+                extension_name = interpret_attr.extension_name,
```

Comment:
The extension name doesn't need to be passed in. It's determined by the `python = module_extension()` line (like a rule name is determined). Just hardcode the extension name to "python"

Reply:
The name is providing us a unique name that we use when we have multiple tool chains.

Reply:
The repos for the toolchains are already unique. Calling python.toolchain() creates 3 types of repos:
  * {name}_<platform> these are the downloaded interpreters
  * {name}_toolchains this is a hub of just toolchain() declration calls, one for each platform, that point back to the the <platform> repos
  * {name} this is a hub like {name}_toolchains, but for some other stuff (I can't remember what)

Reply:
> {name}_<platform> these are the downloaded interpreters 

But they are not downloaded by the toolchain call.  They are only downloaded when we put in the host_python_interpreter call.  I could only see the new messages they downloaded when I added the code for the host_python_interpreter extension.